### DISCUSSION TITLE
Distinguish between Blocks and Entries for Matrix and CK Editor

### DISCUSSION THREAD

**Discussion by maxstrebel** [craftcms/cms#14606] [General]
While working I kept stumbling upon the missing differentiation between blocks and entries. I think to at least in theory distinguish between the two could have benefits. Here some ideas:

## Have an own view for Block Types

Would make it way easier to manage blocks

![Blocks](https://github.com/craftcms/cms/assets/910594/cd8247bb-6210-4fbe-b227-fea819a06f4e)

![Matrix](https://github.com/craftcms/cms/assets/910594/b00223ed-3247-4cc6-8c92-14771a78529e)

## Maybe allow to choose more display options

This way we could choose how we want the fields to be displayed. There are some places where I miss the Super Table style to show fields. Others I want to show as cards.

![Block Detail](https://github.com/craftcms/cms/assets/910594/0b794720-bd90-4c54-9316-dd56b5ab27c0)

**Comment by brandonkelly** [comment]
There is no such thing as “blocks” in Craft 5, nor are there “block types”. And entry types can be shared between sections, Matrix fields, and CKEditor fields. So there’s nothing to distinguish in the first place.

**Reply by maxstrebel** [reply]
Sorry, I guess my discussion was poorly worded and thank you for the answer. I'll just leave some context and let the discussion rest.

I wanted to distinguish between nested entries and related entries. It was inspired through our discussion where you noted that you (rightfully) would like to treat a CK Editor entry that is edited in a slideout the same way as an entry relation edited in a slideout (which should save the entry right away for all instances that reference it).

I am sure I lack most context what led you to the absolute entryfication. While I like the simplicity behind it and would be happy if you could easily move between categories, tags, sections, etc. it does not sit 100% right for me when it comes to matrix and CK. Technically they can be the same (things with fields in them and some meta). But logically and interface wise I think they are not.

|   | CKEditor & Matrix (nested) Entries | Section Entries |
|--------|--------|--------|
| **Entry creation** | Are created inside an section entry via a block editor interface. Type is selected in a list that should help authors to distinguish between content blocks (with text and icons) | Created in the backend as part of a section. Entry type is selected in the sidebar as kind of a template and has different interface needs than CK and matrix |
| **The way they are accessed in the backend** | Accessed as part of a section entry. Should show certain things that are important to see how the content is structured. | Accessed via the backend and search. 80% of the time should show the important things like slugs, title, articleImage, meta, url, postDate, etc. (of course there are exceptions) |
| **The way they are related** | Should be part of the section entry, it's drafts and versions and are not seen as a relation | Should be an own entity and are used as a relation on purpose | 
| **Slugs & Urls** | Are rather needed as an exception | Are usually needed |

**Reply by brandonkelly** [reply]
Yeah that’s a good breakdown. We decided to go all-in on entries for a few reasons:

- Having them all be entries means you can reuse entry types regardless of context.
- While I agree that slugs/URIs (and authors, post dates, etc.) are _usually_ not needed for Matrix blocks, they’ve each been requested from time to time for Matrix blocks. And on the flip side, it’s been requested that they can be _disabled_ for section entries. So it made sense to just have feature toggles for them, rather than have Craft be super opinionated about which features all entries should have, and Matrix blocks shouldn’t.
- Not implemented yet, but having nested blocks be entries makes your content more portable. Down the road you’ll be able to move section entries into a Matrix field, or vise-versa.
- There’s no technical downside to having nested blocks be entries.

From an authoring perspective, it should be pretty much moot, as we continue to make it possible to replace “entry” with more semantic names everywhere (e.g. #14602). Eventually even the “Entries” section in the control panel will be replaced with “Content”.
